### PR TITLE
Add PointCloud datafeature

### PR DESCRIPTION
**Pull request by AhmetNSimsek**
Relevant branch of preconfigured point cloud data for mouse: https://jugit.fz-juelich.de/t.dickscheid/brainscapes-configurations/-/tree/mouseDatasetsFromMaja

Dataset: https://search.kg.ebrains.eu/instances/27dd2588-f4da-406c-9d01-8569bf912b0e

### CODE REVIEW

File: siibra/configuration/factory.py [hunk 1]
```diff
@@ -20,7 +20,8 @@
     receptor_density_fingerprint,
     cell_density_profile,
     layerwise_cell_density,
-    regional_timeseries_activity
+    regional_timeseries_activity,
+    pointcloud as pointcloud
```
Comment (xgui3783) [inline siibra/configuration/factory.py:24]: redundant?

File: siibra/configuration/factory.py [hunk 2]
```diff
@@ -339,6 +340,19 @@ def build_pointset(cls, spec):
             coords = [tuple(c) for c in spec.get("coordinates")]
         return pointset.PointSet(coords, space=space_id)
 
+    @classmethod
+    def build_pointcloud(cls, spec):
+        kwargs = {
+            "modality": spec.get("modality"),
+            "paradigm": spec.get("paradigm"),
```
Comment (xgui3783) [inline siibra/configuration/factory.py:347]: paradigm, datasets, species, do not appear to be in the spec

File: siibra/features/tabular/pointcloud.py [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+            data=None  # lazy loading below
+        )
+        self._decode_func = decode_func
+        self._files = files
+        self._loaders = {
+            subject: requests.HttpRequest(url, func=self._decode_func)
+            for subject, url in files.items()
+        }
+        self.paradigm = paradigm
+        self._pointcloud_cache = dict()
```
Comment (xgui3783) [inline siibra/features/tabular/pointcloud.py:66]: I would add typing to this

File: siibra/features/tabular/pointcloud.py [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+                location=location.WholeBrain(space_id)
+            ),
+            datasets=datasets,
+            data=None  # lazy loading below
+        )
+        self._decode_func = decode_func
+        self._files = files
+        self._loaders = {
+            subject: requests.HttpRequest(url, func=self._decode_func)
+            for subject, url in files.items()
```
Comment (xgui3783) [inline siibra/features/tabular/pointcloud.py:63]: I don't know the answer to this one, but I want to raise the issue:

The term pointcloud seems to be pretty generic. But the implementation seems rather specific. 

my two questions are:

a/ is it worthwhile to subclass pointcloud to create a more specific 
b/ would all point cloud features have subjects

File: siibra/features/tabular/pointcloud.py [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+        values = {}
+        assert array.shape[1] >= 3
+        if array.shape[1] > 3:
+            values = {f"value {n}": [] for n in range(array.shape[0] - 3)}
+        for line in array:
+            coordinates.append(tuple(line[0:3]))
+            if len(values) > 0:
+                for n in range(array.shape[0] - 3):
+                    values[f"value {n}"].append(line[n+3])
+        self._pointcloud_cache[subject] = {"Coordinates": coordinates}
```
Comment (xgui3783) [inline siibra/features/tabular/pointcloud.py:100]: I would use a dataclass rather  than untyped dictionary

File: siibra/features/tabular/pointcloud.py [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+        if values:
+            self._pointcloud_cache[subject].extend(values)
```
Comment (xgui3783) [inline siibra/features/tabular/pointcloud.py:102]: I think there is a short cut here:

```python
self._pointcloud_cache[subject] = {"Coordinates": coordinates, **values}
```

File: siibra/features/tabular/pointcloud.py [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+        -------
+        PointSet
+
+        Raises
+        ------
+        TypeError
+            If no subject is specified.
+        """
+        if (subject is None) and (len(self.subjects) > 1):
+                raise TypeError("Missing argument: 'subject'")
```
Comment (xgui3783) [inline siibra/features/tabular/pointcloud.py:161]: indent looks a little weird

**Review by xgui3783 — COMMENTED** [review]
In general implementation looks solid. 

My uncertainty is more architectural / code organisation one, that if `PointCloud` is ... too vague for our use case.

File: siibra/configuration/factory.py (hunk 1 above)
Comment (AhmetNSimsek) [reply siibra/configuration/factory.py:24]: Ah yes... it was called pointset before so I forgot to remove it and I was calling as PointSetFeature

File: siibra/features/tabular/pointcloud.py (hunk 4 above)
Comment (AhmetNSimsek) [reply siibra/features/tabular/pointcloud.py:63]: Hmm. I was trying to make it as generic as possible. But I see what you mean.
I will think about it a bit and check other point cloud data on KG.

**Comment by AhmetNSimsek** [comment]
The dataset used for testing has changed today and the files are inaccessible. Using the new files results in "internal server error".

**Comment by AhmetNSimsek** [comment]
This PR will be closed in favor of https://github.com/FZJ-INM1-BDA/siibra-python/pull/399 which provides a more general approach.

**Pull request closed without merging**